src/cursor/cursor.ts

```diff
@@ -828,10 +907,7 @@ export class Cursor<
         this.kill();
```

Comment:
just a thought since we're here: `this.kill()` sends a command to the server, I think we might need to await the result of that process (checking connection out of the pool, sending command, etc) even though the `killCursors` operation is sent not expecting a response. Can you take a look, if you concur can you make a ticket for this and add it to the epic?